python/Ice/async/client.py

```diff
@@ -22,62 +22,51 @@ async def main():
             return 1
 
         try:
-            hello = Demo.HelloPrx.checkedCast(communicator.propertyToProxy('Hello.Proxy'))
+            hello = Demo.HelloPrx.uncheckedCast(communicator.propertyToProxy('Hello.Proxy'))
         except Ice.Exception as ex:
             print("invalid proxy:", ex)
             return 1
 
         menu()
 
-        tasks = set()
-        done = set()
-
-        while True:
-            sys.stdout.write("==> ")
-            sys.stdout.flush()
-            nextCharTask = asyncio.create_task(getNextChar())
-            tasks.add(nextCharTask)
-            while nextCharTask not in done:
-                done, tasks = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-                if nextCharTask in done:
-                    c = nextCharTask.result()
-                    if c == 'i':
-                        # create asyncio future from Ice future
-                        tasks.add(Ice.wrap_future(hello.sayHelloAsync(0)))
-                    elif c == 'd':
-                        # create asyncio task from coroutine returned by slowSayHello
-                        tasks.add(asyncio.create_task(slowSayHello(hello)))
-                    elif c == 's':
-                        tasks.add(Ice.wrap_future(hello.shutdownAsync()))
-                    elif c == 'x':
-                        # cancel all outstanding tasks
-                        for t in tasks:
-                            t.cancel()
-                        return 0
-                    elif c == '?':
-                        menu()
-                    else:
-                        print(f"unknown command {c!r}")
-                        menu()
-                # retrieve exceptions of done tasks, if any
-                for t in done:
-                    try:
-                        t.result()
-                    except Ice.Exception as ex:
-                        print("task exception:", ex)
-                    except RuntimeError as ex:
-                        print("task runtime error:", ex)
-
-
-async def getNextChar():
-    # run blocking IO with default executor
-    c = await asyncio.get_running_loop().run_in_executor(None, sys.stdin.readline)
-    return c.strip()
+        c = None
+        while c != 'x':
+            try:
+                sys.stdout.write("==> ")
```

Comment:
Should we instead
```
print("==> ", end='', flush=True)
```
?

Reply:
Either if fine with me as long as it still works with our expect demo scripts. If we decide to change this, it's best to do this for all demos separately from this PR.